_posts/2020-04-21-0.17.0-release.md

```diff
@@ -0,0 +1,197 @@
+author: pmc
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+
+The Apache Arrow team is pleased to announce the 0.17.0 release. This covers
+over 2 months of development work and includes [**569 resolved issues**][1]
+from [**79 distinct contributors**][2]. See the Install Page to learn how to
+get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Many other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## Community
+
+Since the 0.16.0 release, two committers have joined the Project Management
+Committee (PMC):
+
+* [Neal Richardson][4]
+* [François Saint-Jacques][5]
+
+Thank you for all your contributions!
+
+## Columnar Format Notes
+
+A [C-level Data Interface][6] was designed to ease data sharing inside a single process. It allows different runtimes or libraries to share Arrow data using a well-known binary layout and metadata representation, without any copies. The C++ library now includes an implementation of this specification, and Python and R have bindings to that implementation.
+
+## Arrow Flight RPC notes
+
+* Adopted new DoExchange bi-directional data RPC
+* ListFlights supports being passed a Criteria argument in Java/C++/Python. This allows applications to search for flights satisfying a given query.
+* Custom metadata can be attached to errors that the server sends to the client, which can be used to encode richer application-specific information.
+* A number of minor bugs were fixed, including proper handling of empty null arrays in Java and round-tripping of certain Arrow status codes in C++/Python.
+
+## C++ notes
+
+### Feather V2
+
+The "Feather V2" format based on the Arrow IPC file format was developed.
+Feather v2 features full support for all Arrow data types,
+and resolves the 2GB per-column limitation for large amounts of string data that
+the [original Feather implementation][13] had.
+Feather V2 also introduces experimental IPC message compression using LZ4 frame format or ZSTD. This will be formalized later in the Arrow format.
+
+### C++ Datasets
+
+TODO: SUMMARIZE C++ DATASETS WORK
```

Comment:
@fsaintjacques

Reply:
* Add the ability to list files in FileSystemDataset
* Add options to ParquetFileFormat
* Support dictionary columns in partition expression
* Relax discovery validation and improve speed on high latency file system
* Support columns projection with IpcFileFormat
* Fix various crashes and other issues